surveillance-card.js

```diff
@@ -1,31 +1,199 @@
+    this._hass = hass;
+    this.updateCameras();
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.thumbUpdater = setInterval(
+      () => this.updateCameras(),
+      this.thumbInterval
+    );
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    clearInterval(this.thumbUpdater);
+  }
+
+  setConfig(config) {
+    if (!config.cameras) {
+      throw new Error("You need to define cameras");
+    }
+
+    this.focusOnMotion = config.focus_motion !== false;
+    this.thumbInterval = (config.thumb_interval || 10.0) * 1000;
+    this.updateInterval = config.update_interval || 1.0;
+
+    const now = Date.now();
+    this.cameras = config.cameras.map((camera) => {
+      const entity = this.hass && hass.states[camera.entity];
+      const attributes = entity && entity.attributes;
+      return {
+        access_token: attributes && attributes.access_token,
+        entity: camera.entity,
+        motion_entity: camera.motion_entity,
+        name: attributes && attributes.friendly_name,
+        has_motion:
+          this.hass && this.hass.states[camera.motion_entity].state === "on",
+        last_motion: now,
+        last_update: now,
+        stream_url: "",
+        url: attributes && attributes.entity_picture,
+      };
+    });
+    this.updateCameras = this.throttle(
+      () => this._updateCameras(),
+      this.thumbInterval
+    );
+    this._updateSelectedCamera();
+  }
+
+  _updateCameras() {
+    const now = Date.now();
+    const { states } = this.hass;
+    const activatedCameras = [];
+
+    for (const camera of this.cameras) {
+      const hadMotion = camera.has_motion === true;
+      const { motion_entity } = camera;
+      camera.has_motion =
+        motion_entity in states && states[motion_entity].state === "on";
+      if (camera.has_motion) {
+        camera.last_motion = now;
+      }
+
+      const motionActivated = !hadMotion && camera.has_motion;
+      if (motionActivated) {
+        activatedCameras.push(camera);
+      }
+
+      // update if there was just motion occurred or the thumb interval was reached.
+      if (motionActivated || now - camera.last_update > this.thumbInterval) {
+        camera.last_update = now;
+      }
+
+      camera.stream_url = `/api/camera_proxy_stream/${camera.entity}?token=${camera.access_token}&interval=${this.updateInterval}`;
+      const attributes =
+        camera.entity in states ? states[camera.entity].attributes || {} : {};
+      camera.access_token = attributes.access_token;
+      camera.name = attributes.friendly_name;
+      camera.url = `${attributes.entity_picture}&last_update=${camera.last_update}`;
+    }
+
+    if (this.focusOnMotion && activatedCameras.length > 0) {
+      this._updateSelectedCamera(
+        activatedCameras.find((c) => c === this.selectedCamera) ||
+          activatedCameras[0]
+      );
+    }
+
+    this.cameras.sort(this._cameraSortComparer);
+    this.cameras = [...this.cameras];
```

Comment:
sort mutates but only if changes so I had to trigger a refresh for lit element so render happens.